--- Day 1: Trebuchet?! ---
Something is wrong with global snow production, and you've been selected to take a look. The Elves have even given you a map; on it, they've used stars to mark the top fifty locations that are likely to be having problems.

You've been doing this long enough to know that to restore snow operations, you need to check all fifty stars by December 25th.

Collect stars by solving puzzles. Two puzzles will be made available on each day in the Advent calendar; the second puzzle is unlocked when you complete the first. Each puzzle grants one star. Good luck!

You try to ask why they can't just use a weather machine ("not powerful enough") and where they're even sending you ("the sky") and why your map looks mostly blank ("you sure ask a lot of questions") and hang on did you just say the sky ("of course, where do you think snow comes from") when you realize that the Elves are already loading you into a trebuchet ("please hold still, we need to strap you in").

As they're making the final adjustments, they discover that their calibration document (your puzzle input) has been amended by a very young Elf who was apparently just excited to show off her art skills. Consequently, the Elves are having trouble reading the values on the document.

The newly-improved calibration document consists of lines of text; each line originally contained a specific calibration value that the Elves now need to recover. On each line, the calibration value can be found by combining the first digit and the last digit (in that order) to form a single two-digit number.

For example:

1abc2
pqr3stu8vwx
a1b2c3d4e5f
treb7uchet
In this example, the calibration values of these four lines are 12, 38, 15, and 77. Adding these together produces 142.

Consider your entire calibration document. What is the sum of all of the calibration values?

In [1]:
# input

personal_puzzle_input =  """2xjzgsjzfhzhm1
qhklfjd39rpjxhqtftwopfvrrj2eight
95btwo
lfsqldnf3onenplgfxdjzjjnpzfxnineseven
five7fouronesevenpfsrmszpc
xcsfkjqvln2tpm
ncjtcdb625two9
6srtj3gfghkdtmfzseven
three4fournineone
pcpsxmkhdgsix25six6ngzsqzdcrb
1sixdzhbjvnvgcxmsrlkb9xbjk7t2
dsxnfkjn2vtwofivethree2
qfbp2tpgvsxm
kvbhsseven2rkbllhrhvsevenfour2vf
bqbtzbtt3
ninek6
3fivebvqhvhlvz
two68jxfnqlzfkninenine8
vhckhhhb2sixtftjtdjf2nine
4jc
3sevenxlxfnpk
5sjfive
c8bvlkdq283rqfftj
8eight1threehhvmnlft
4sixmfcronelzcpnfourlgdbkgt
hntcjmppp9fmntwokx2
24mkgr6mx9n4
cksfkrdmdcmnbmkgk3two4
twosixthreefivelq3nine
njmrtr6fmldninevcbtxr42
eightfrdrczdxrcrfxh6
8sevenonefmfqhtx8lk
lnoneight8twodg
tszrcb7tvvhfive4
nknphprtv67fivehqlfournine2jk
9eightszgdhftggrktkzbsmnhtwonekh
1onefzkgf2two417
77fivepsmmpp
bnine7vmttgmdskc
6tnsixclbvkhmsp
ggreightwo1mnsbzmxsdcdrhhg5one
ninetwo6qtbklgb2
four8fmfxmr
threetxfrsevenvfdgrtsixtwofsq7
hqxlseven53
vxqq24eight6
7psxpgxmkpzkzeight3
9four7
stwoone4eightwoj
one38
9cjlrgrgpfjeighteightdcpjhzbjqcdthree
rqfone2
dvkxhvkgxcfivescjcnqqnnine2tthreecptkgxmzjsbzdrhk
5lhprmg8two7two3
5lvtg5sevennine2zrrmghf4seven
fourfive97sevensevensixcsffnlcbtjk
onesixoneninesixnbkf5xpsgsmpvdsdsqdvpjjzp
nineb321mgkpqjqjqh
seven2cstttkvkmpssflkxjmbnvthreefour
six1bmcjbhdgqm6kgzsxbgdqnjqf3four6
zchkfjxsllkncz175
vsveightwo8
fourbtrnfdp2four5
13dtf8769
6qfvvdqdrtstwothree4seven8vszpseven
7rgzpxfbczk1fivetwo4two9nine
3sksmfxn
16ninetwonqqmptq
9five4fckzssxsvpzbvlktjzcninethree
3threel889bvbcgvl
threetzzjntxlsd2klrjfnbg87
3two3qrrdbkbk
29oneightt
7mkpfchstjt7mshqht4fivesix
tvmqndvsix875nine8
vg7477ninecpnrvnine7
seveneightmknlqjlstx4zjjrpbhjhnnc8
8qklthreetwo7tlmrfzvtwo7zvccc
6one1dpstqpmfbhst
sixfourfourseven6rzdkfour
6eighttbltmntb8
seven72cqslvzpgj
jpvrsgfhtwo5nfc1cgxdrdrbfnseven
four1ninezlhqxtsgfzsevenhspvmxrtztzgtmseven5
foursixzjvgjgsevensvkd5
seven121fivefgxhdfive
twoh8
ghhtttxqgr95
zkpklcjbjlr2
nine2mtrdcbrbntrdqninevbkbfg35lnpx
nineninesix68pcvsdnns
rf16one7
seven276
sixseven1
qjxseven9six
7c9xvhgmpf41
dlzpvdpxseven9nine
mlmlqrpn4twofivesbmhdbcsixtwo
ld74txfzksr5qconenine
six7djcdrtk7chktrh88
2seven7jtrbhznt
1eighthnhchsevenfive
qdhscpeight8
12gnxzzm
k7fourmhcfkggt8919np
8zljspvnmlx6four
36twofivethreeptbdrfqzbz7qbjm
bxkbjbzgone3ldhlnhxfcf7btkktspxrtqkxqfdmlqgqvgclb
sevenfourmdrxseven2z2fiveone
8bbprzqrqn
twofivethree95mpqclhfkzlsix
gpfjflrkc96tmsix6dxjnfive
qzeight6
five5zlthgqjntwotqmnnsd
eight6nbpvfour4kzkvbzrxcmkh
fdkrxdckqbpsklz64fourones8
5qpldkhltl
27xldkbqbnmrrbqkhmksvk
pjoneightknpcgkkv7cbknrhfdmtdm8
432xsixvvktflzxone
35fouronemzzszfqppllgchsjjnine
qgcnzcbvsxbtn6lthreenineone
two2fourkdgbfb
fzkmbtvnvj5bzfgvxv3nineknpqzznzmf
9mkqtmlnpzt282dgkmvlsonefourthree
mcqhpkhonethreeninesevenfdsggqnlfcnrft3frqzkscft
847
gsxvvr6tghfbxxghgv6lplf
fqzgkxkbmkpbhpvpdprmlhsngc5
qeightwo2threecrjjtv9fourbthree
2eightdcbgfdqzk
fourtwothree5tdzpzfxkpjmlkzzrf
seven633
8mldn
eightmg7dvjzbngkxmnxdctjtdffive97zpjrjqvp
sixfournineone38eighteight
8mngs412
3fivefourtwo7five
7sevenjfkcf7rvone8xjdvsc
tljcrzngfdsqctfour6hsnljsxrqdzd
csix7hnkhds4four67
3cxrlqcnseven4twox3
rtwonerxcnpqdnzksvzmmndcjlzq2nine6ntfzjdqnrjeight
zbqgrvvvblggbghtfx98
three2zcpspgtq77vqtgpcxkvq8
94zksixnvbxrllxln
1eightrctknxkmv9jtwoeight
2onenine5eight58
four2five
6fourfiveeight
pbfourlzdl4
4onejrltvhxprhlzctrqzbz5fourseven6
hlsdjlleightdcmdj8jdg
eightbfkgvbxxm6
sixseven9hmhchsmqhhtrdftcfivetz
mq4five2fsdhzn
ceight3mncgnineeight8dkqrjjtmc
onefour7eightsix
seven4fqsjk38fhbmn9fgzzcc
9cmb5ninenjsppthpnine5
qghtvpsqpxptoneljvstwo8lpqcbgtqvjmpq
onelq85dlqpx8
1threetzsxqtndvhqn
lqdvkonebqjqj9
3six6
9two9sixeight776
scrrprfkblkplgqzzqb2
18vcrhxb4b
three8nine
1khkncj8fivefourthree2
ninetwo8tbbsfmkqtt51five
66one88
ninempgbhpm6seven34
eightthreetwo4threetwo
vvpqkdonesevendvfrgmhhlsix91threethree
vbskcncpjxtdhqlcone6kgqpz
54lmnnqsrm73
5mqnmvcbqnz82ninethree
47teightthreehrhtdndrk
fft2foursrdhxkzfourv
tbeight48759eight9twonenl
3sfkvkmvbbm9four9cbjzgllfl
sixqhxeightone7tlsd
22zmllbhcptwo
dqxdjrkmnvkbsxc3ztbseven
oneeight6dcphmnrfltfour
mgxzcfsevenfour4six
sixb8six34gqjtlcq
fiveseventhreexfblvxgkplvbrgpfive1onesix
one3mgqphj5
7fpzxmmxgnb5
jbshclzone3oneseven117dkzzdxdd
56five1
jeightwo1ninemxtvgl717pbv1mqs
3six36
35mlzjsgxxtg
hmmjgonehjqdcqkh1gskjrmfour7
eightfiverqbb3
sixtdkfbglhbjbvthreejsd8tglkrsfktcjt
1eight2mgxrgbbb3
zzqxvtthreepf3eighttwoonefive
vjnxhl4threetwokj
oneqddgbbqtzsonenxkeightvx7
nine24
ftwone2
53twonehx
two7three4fourhbgninenkvxjjsz
7five62ninefour
53nine7lqthree4
1six1jn
five5drcjvbgdxzonefourfjdkfive
6onecfninesqfdtxmdhmfvlhbsjv9eight
5dqmfmqbdfdjvpl
qxkqfjshdone2zrmdsts
1qrvnf8zvfour
dht2
4twoqdrshn87vxldzvnkn1five
4nineh
zctqdcfbseven6three58lzscrpsbb2
fzmndhssmm3jdzlfourshjkmjkv53gvssnvzsz
five2qkxtfkj3sixshpnzdcjhlcmksnv
bskcmljnq69
psxkxqztb3xsctsk6qvldjdzzqmgqcxncx2btqxh
onetwosixttghndtzx8
8twocmnfmslsheighthtxsgjddm
fnvkqndfqmhgscnvq31kvvftxbzhtgzxqtngvqseven
7sevenkhhvtvsbrvcvm8zksvgcfrkgfourtzjvpqstckmmlzz
7hkfiveeightpjhone
8zptwoonemrninefive3spxrlhr
3qvjknqcnngpkninecgsbp3sevenfnpvrcltj
qztwo8678
bxkj959dntcpzv
4one4
lstwone347nine
ff79krfbjttqp4seven1
gjtlbgxdqt7eightsix
8fivefour82threetwo
1zjxthree4
szgx6fivefour12
557dspnfzhmlh
5hnfour
847zzkrtjlb
82six
xlnhrcksv3fourzl
2ninetwo8ninelhnsbseven9
ccseightwo8fiveonetwo5onegkhlg
eightp2onegrmfsnine5one
4five9xnmlldfive
five636
fivesix29
34dsbtmlhgggzkqxs7
twothreesmgjch93mcdndsn
cmnmtzjnmtggseven1tjsshnktffive72
7xcbpsix6m
7fourthree
7cxnzthree
5eightthree3lpvlfk
4rdnp
one6gmpjfouronettbbjdqsxn
6xtcv
dpspd1cb2onegxsgdtcsix
1eightpseven
ntczqmrclm2
six7nineeight1
qrndlfour9bhbmlgj6
52vhnxvngjjkzt
nine8six
3seventhreesixfpmpbjninesix
seightwoseven4sevenppdnk7
lnxeightbhxfmdrxcbjzx9sixnine
8ptdmjrd21zrnt1one
one1dmbfournine2
25five748
266eightwom
zxfxjn32threelxmqnlcvnknzbn3
lqmjm2onezseven
sphlhbbz14sixtwosix
sixr9llqxb4
tnnhlmsix314mqvdlxfour4
threenjxgtthreethreeeight14one
1fivetwo
hhhtkpvhvg8
31cslxxzgbktjtwotrzs1one
twoseven9
4fivenine9332
six35nineeightwok
nine9fpxgqjrftwovhldfflqthree
434four5threetpcs
ql52kzzgnsgll
onex5145
gsppnkx1
22six
zjbbqnbhsdfourmk19sixone
six3onesevensixvnsixone7
seventhreefive8
four62four7eightseventwo
2three5jtzmtxhqxhsixxlkmptshb
five3lklnccst1dstgvnprktgcnseven
sevenfour52
hfxkkbcfpcgzfqsv1three68three
lnqrcdpglf2ksqddhp
13onenbmqvpqsbbf
1bfive
1dmnlscgvfdqrf
bvtwonettkzzxpb4
5two2t7
t7djkh
63t86nlhfgqgnrghqone1
one89eightlmztkcjjnonefive
7twosjmcrlcfqr29lfkrjz
1nine5four
xxjbmsvfxb8
nxvrvcqkbseven99rbzbdp641
56fdpx
skszhkqdtonesixth5eight
ninetwo8jqbtwokbhzvv4fiveoneightxb
1threemfcblxjljdtone
fsfgncxvcmszn2pzspkkgqvfxlnbzgjr1
rrxhnslrkeightgcsbcgdbkplzxbmmlxsn5sfp66
foureight3
5rplrmrjtjcfnct9sixfourzptmffrpthree
mld6ztzninesixfour4
twohbbxmthljpkqthtcgtt8
ltvgmqgsix89
6khclqhcvdz24eightseven
6prb
nhsjqtnk6bqzhzvldlcgsmkq1twobfqzm
tfivehmdtxqlsevenzxt1ckjbnttgqctwo4
ghnjjmtlvfive9znltjfskfnl3vdcjkvqcdplvdmckmfgxq
ztvvbbjx3
6jhsrszlpsh
4fhtrctvrgmfhdjqvqp8sixnnftggrzlgs2
bvlqxnjbbjgtfpslpspvpl7jsjlrdfiveeight88
fsvcxvlc74sevenltx37sevenlbq
fourfivetgkqz5cgj4
cnrjtfvht74nptvh
oneldnoneone745dhvqnvmlfj
2trhmdcpnreight
dshghlkjzh1kv
threexzkssrxlzzpxsjzbbtthree7
eightdsqrdllssxv7sevendttwo3three
stz6five9four
jfbfgvfgmvlhjv94
rcncncfcqsevenjdrtctbbsznpct48cbbxseven
ninevjxzk2two
1qgdcpslvqhjzbgtsbnzcd
kmlzbndds8fivegfdgpfivetwo63
66mkeightnine1rvfnine4
nine82
58ntpkdssixnntjlgfkxq
2jxqttctsxthreepnxsfive
fivekvfkqpc28foursixptgbpfm6
6twofourslqbzlcsmf
five4vmdl
thx5six
5hfnsqhvcfgckcseven3jgdtkkv
seven6eight65five99
hplrgdqrjtonesevenseven4z65eight
khkts8hcsfvhvqjv9eightnine48
pqxztthreeninelxhcxkjclk2
sixrmfpj6nine4zx
9xtvxkxpjv9
cmtggbrckmmzffncttxsevenhphmhfgjsr85
83s4
26fivefivethreemstzpncmhfm
87eightfourgvmmfbknineone
bdxg2zszhcgtvtseventhmbvfive
zeight8
eight78dcnq4jtz
67csrthree
3vqxtwonerts
1five3
sevensixfournineksxkz9bkbgcrjjjhnineeight
8seven29
9394
pkqxvl9vbjmqblrksjqcshpzpfrfouroneeightnine
heightwolgpplrbnhrgdkrk25crttkkzkbt
6659eighttwoone
55lzqzng2dmtpq45
512gpmqpsqfive9eight
eightsnxz58zrbh8fiveeight
six893ninerrxzj
twoflrzzcgninefqqlmmzzqgnjrcxcfour5ninetlcmc
tmfkbgcssdcmkthmx1seven
6three9fourmdpzj
mgprpn68
xssxrtpntlkppxp958rzd2nine8
fx15twoeighttfjlcglnbptwo66
threeseven522
xrjpstjr2
6eight1742oneighthhj
1ddfkgjrxfzrhnxtnvfqxdv4
8hlcxdpkrnrgfzmsschlhcbndkcvdszgqntl
five154sixfourqgffh
5rsvtzgrvq1slrlgqksone6lns
92sevenk3xfrvgddsthree4
rctlcz67
2nmrbhnzpf7
bjmssix1sevenfour4
744ninefivefivejjcnpgrqxtrctwo
r4five4
16two2
threefn8
3gmsrqtjm
one8tmlnlqvgfone5fivefive
1spllmone9three1xsssk5
6rlqqblbrxh8
41qqccnsevengbbrpgcqggtlhtzqcfqzhtx
hzfivethreenine2
sixeightvcm4lngvkdlpz6dstz
twoeight34tv1
tgqc6fqsftwoninexncjnrtcgcbrxdmnine
8one27ninerjt3seven
1sjvxlzszhhsix
2xxmpjsfbr
7eight5eight5bzgdpkkpgrn
twozcrxdsix1lrxltrhvznine7twoone
8qlfive8nlmthree39
4five5five7671
4fiveqtrngqtbpd74
vbccfjhjndeight1h
8kfzchkdvfgvlhqzfgseven
6ninexrxsvlmmzrsevenjhzzggfcxqrvfjtnjctveight9
6fourfourfourxgrgqklnpcsjbsqvhnkjfcjdhdf
65fpnfszb3
srnhthmpseven1four3rrpznf
vgmqr9zttpdrzjfiveninesevendfive
fournthreesevenseven72seven
9594
twolhvpfzbfsfourrrcpzzshd66grxddrmkvthree
seven5twotwolpsqlbcqpthree
29eightzjgsfmll
gtkrjrsevenfbjcpjlsmqsfceightseven9
qkbntcjvfl6three9qhpcptvpkfour7b
gvkhmrljmhrpj956bzxhkljpvpxltctjvvpfjtsnb
rcqlnpcmng6l5vfkpddeightsevenfhmxzztndq
sptdrlsfxpsxff4five
lrzhhxdrkdtbdjbsfourthreemmjjnn1
9sxthreethree
six9fourseven4
dpone9fshbbpcm8
twoczqzvqkcnine6eight
qbtkgxp1fourdz
dxrrsdbmtpnl6hfdtj
sixsvtwo8
5fxdkhbgqm8fouronefive5
jqj6lhtbjt
6pdgrsdxghfkhptbjcrrjcsevenbksvmzvnninercz
8rzqvh6ntlg2bfktcphztdkckjg7two
fivesixnine1
4beightsevenonetwothree8jx
sixfourmvpk6fourtgknrjmxfftnmngjnine9
68seventwosnqsj
g853seven7
7ninejqslxmkxpbkxksjcjlnttslpcmsprvgnrxg
jmvqrhdcctwofivev6eightone
ninefiveseven9eighthbb7two
threethreethreerlgqzvsevenx3eight
xbkvgdsrhknine3blt
xzbcgjkkbk1two1zprpn3bcxxhlxjtx7eight
3tflfncljtwo9
four3seven8one6
six7vqfive
ddhhgvrkzfhone8
fqtbnbvdvbklnkmpr89twotwo4
9two166eight4cfnsstwo
twofive88
nxpmbfiveeightqmtblmtzpjcnxnxjtwosixtwo9
fourthreefivedone9jzhrbrzmdqkzmqzfklxq
qbgpfsvlx1eightjsevensixfour
2sevenfqkeight2five
sqnsvsevenrbkhgkbhmmqjztzvn3sevennkcghqqcckp
eighthcrzxpjt1eightone
sixone2
four56zqxr1lcmblzgdgqjnqb
q6hdldzmjh6ninevjm9
j82onek67
5njnffiveshlbqpqhzt
97nineoneightcq
ffivetwo41vnzhfqmrd4
two2fiveone1nftfhvzkpltvn4three
twofivethree7jlsjxgjfbvnlxhb
rmb4fourlmzgvfcsgn2five
6dthkxkcdbssixdlrb38
qgbnpdpchmnnftqpeight9xlmlskqhc156drhpf
sevensixsevenjqckcmg6sixxtgvrvm2sgtvrjrq
4nnhmghrqrkcmfmqxtqxncljskfvmhfmgqfour
dlpqjk2fivesix7
onegtddk5vzgpndqp2five
qsslgone4onelbnbckznrfnjmsgz
pvqdbhjgqrgbqkzgpdbmvninekhvfrmpchhxm2
tpxvskgzgnnvmnfncvqxxtkmm75vfgmxr4zbpdmdseven
ssnqndqxsevenxztmfq64
mcxmhtnbczdfnstjn31six
twosixfz5jvbxxlhdklbp1
five4six952
vmnhjm3rzgglnmzj318
dcnprh12qnmmqnvhkrhtrnxs5
5b
mkfgxmvsixeight5
crzdkbjthreeone287fourpgzqldn
2cgdd
seven8oneeightllqtfbdtmdnvlcsfh5
4seven1zqponeeightlplm
c7sevendvmxlfpnr6
eightlkhjgqj6
njvmgjdmx2
9six7
3cfbjptcp5one8four
eight6kxc8one
6sevenlpbdcgbhxsixfourseven7
4sixeightsix9hq
6eightb
8eight86jkrdgmxdfourthree
4dxczptzbvjfive56eight
seven3threerpqhsqspsdpvxj2sevensndcxn
zqcbcseventwo4gh16jksxkphgf
44tfour7br
five266five93seven1
prsbtlbninendbpzv5xx5
6three2nqgtrlkfxn
ninefourxkrsgshmdd78six7nine
gnkphdvhqk74fptcqgp
cfdmp2
fzhhdclzncxmg26nineeight4
seventworlcqhfq2qrfiveqqgmmzdpntfm
onedvsbvstdshtlzs6sixfive
qjfbtdtcknthreeeightsix71
2ninesixscdxtwo
threerlpjxlzsr9qqzmfbzdtnknn2four
8kxxljngrqsgkxcnlhrlxxfgl57fql7
vhrjlgqjnnthree4four8fivebfourffcnbghnz
5jb5oneoneonebseven
kbkv3three4rgdkdzkrkk
eightxhxzkbhjxsvfsjfbzghktvlqhseven2xkslcrfsgdc
sevenone6rmkcvfour6jhvhz
threenhfprcngzhcqckchphcninecmqvczqt5
foureight6
7fkrgmdrxgdkfpvxkns9
5onetwoqnp
nine73dbfljgnninecdgxrgkdlsqzflbtwo
five3jcsevenfour8sevenjgfxkfgnlzbkb
lqtvkrfour67gdnrhkfvzgfoureightone
kct1pljt8seven
six269two
64k
41ggbblvhs726
mbvfq5gghsxtwomngfczfdpb1seven
4ggzmdpbkrczhmmrk5
317
pxtgbfnfhqgncrsgtf7sixfour
dvqnfourcsnmcxnxq7ghmjbgmct
vnskfrdcjdoneddjpkcj7fourjjlllonepxlpjkzbzfour
hbdjzfbvrthreekxmlkm49
trptone5bpsxgl
5one1
sixsixfzgbninecxssl9
six7xdnsdgsix4
fivefive84fxrzgk4eight81
zszvqhms8sstxvfz
3one3sixbmxbfsdbdtmh
qnfgnjgcv6two57
5676xzlbzk485seven
five8dbbtpxctf
two1gntvjr
77one
vlnkqvczts5tkghpvthree6krgszgx4
nine2threefoureight
foursevenhc3twofour
9zzgqfzmfpsfn
ghb57sixeightwofqc
6twovninejckdhmcsxkfoursix5
oneqxljxjhnineppln419
one2zlfsxrkfourthreexl7
nqbtczpblq856one6
3cxmmzlh1
frjsdskzk79lcxvjqqdnlqsmbpbdxrxthreeseven
twoone457fivetbq4
fourtvbdksxsjxzgdk77
fourgbvdlpbgztwoftmkjkkmthreesix1
nhrqfbzggrskfgmnbg7sevenfzrnlxfbtc
foursevenmdrnfqsn4jvnrntwo29
fivevzvszgqvqp6627zqnk
15fcvczdlqprxjzhgbxpxs5
9fourfivejsgvnf9five
kzlndjrn8rgq
spfqqx3sevenoneldqnqbmspvsixfiveninezmvks
oneseven5
2eightfour5bqbmsnhmtfivebss
ninejnckkvseven4
jzhfcmvb8fiveqgq18fivevlpgdnkbq
67four
grzpdtl9
lzf8fivedr2fivevbpldcone
nine4bdfptdffgkssltwopl
eightlvgfhfbsqrdjrskqkprdlltnzzpfhnczsonetwo5eight
7xqfgzglblftvsqqbgpfmthreesix9jfl2
eightgngsvqvrhgmvhtqhbpdeightfour6
1bqvpnggbdrfoureightxmclfpfqgjssrpflnzl
eightfive1qvhhthk9ninetwo
six36fourfivefivezsvndsl5four
jnpzplgbqf6tlzpsgvzhtnkdvhzhbvhps5
cldz3bfpvdxjmc
dpponeseven48sevennineeightrkn
five3fournine
2xcckljnjz98mjjscfjsxsevenqhxrqvqmbvhzgkc
sixtwo66469
79four5eight
jxqktt47
eight36
rsjlmlj5oneightc
xczzpjknphtspmh7eightvfzjjk2five
kmznfcxhsn8fivethreexnineone2
four6six
2hhnine627
eightnine1vkdpgvxgm
8three6pvtqmfdxdkxps
7frnthree
78tworhldxeighttphp
dcmv6vvfbzjdtphbpssjnnqdmrgzgkfdveightwov
zsbmctdseven1snxmqxzgsix9
lxtwo4sevenxsix
tkdkqfpj5eightq8two447
seven1onesixfivest
45six5sevenseven
jpxxdjphzskpbrllninexmmmdxrznzthreecvsmjssmfxczhm5
4flhfqlphfcnn9sevenx3
16ttrmgqninefsgvkt5four
3snfkm
6threetwoglgkbqjlsn4fivehcgfour
ninesixfive1ninelgdmhkqrvn
two3pxvlsheight
two51
sixninezkbfbxgzznlgtpxgcpzvgkczdbfive8three
fourjljzqgbsthreeqjdgsqpspnsixxnglmkss5m
32sevenctqtqrqlsfourfoursixrcjztxx
8qlccjngpt
2eight58five
1ninesevenxtlbrq6
9fivevkmm8tgxfkzvgq
53txg
3xqmjvxkcbjf2
2fhndgqnqd5
2vnrpjdb6four79
7four5
foursix5hdcg489ninerncnk
6s2jd
fournine8ninejn6rkknh5
fcfxflzv7seven7
1fournineqzknrvfhdxfoureight4seveneightwobr
1two3
twombl5
fourzjh9six
1ljrvbrqpcchb
11rvqfmcxgg
fiveoneggjpmdt2t
nnjdzhdrkvhpxnprlfrhxpbtpbdxm1xfshdr
five39threehhrhsrhhqkjt9
hm8nine
3ldckgksm48xc2seven8gbzfq
sixqpcqjzqxsjhr7
kthreethreelmjfour485
mnggrsbbnh1
xfn5
qzlxtczqlvninekjnbnonefcbqqgfmqf66xpvjgfzjgxvjrcqmzx
krsfvctzfhonejxtqbbnd7eight
tjhgbtqtwoljone9
mdfntzfpg1fiveninecvxrfnine2
7hcrxzc
6two6plgrmbnine9sdbvgtnj
threenine4fthstfjzzp
6eighteight9npghczggthreectqssdtp
flc4gtqsjmrddx8eightseven
threeone1oneone7
gfbfjzltcjfkkrsprgtpfive4
sixrdzp2onesevenrqrtldnq4c
sevenktwo3threeeightthreenhnmpdmlqjnvl
4eight82717
nrhqqkmkeightthreenzqbgcphoneseven7
twoprrgggznzffour5bbrdsvtgkkrbl5nlrkpjmnbzgsbtl
28pvjrszmrdntp4nqdnhkdrtwoxr
sixsix9five4
lxchg91threethree74
6bmxblxdkghqdjqgqbrqtxknvdsevenztgspsfh83
39bsspvhslzkmrnzxtlxsbhk4
74seveneight9ninedgzgjdvxgvrnfz
fourxvvqsfktptjbsb69
rxsevenfive5vx9
72six7
five5lcxv2jvsssxqjtmzclfdfourfive
87three9
6rckksfour413331
33gtz
xkmxtvzhgxqqmknn9pzhgjxqseight
8threefivedrtlgrnqfouroneq
6snzkxjcfive62dqtqmtwoxnine
zxtjclonefour1eight5eight1
qlffthreeseven2three
xfppngbg5six3two8nine
ninesevensevencfhppvhdnfour8
xhvnm1
9threecbdccktrkfourfour9twoneqd
1five4fivenine
jcscpdtzk757nine8
sevenninesixxkzzzbfour2three
sixthreetwovfqxtbvcrrjtwo3four9
nzdjhtlqfnxmrzvgrzkfmm7mhmvbbxfd
3seven2eight99
two6nineljvxs5eightseven3
55fivezgbmcdpv
hkgvvrtrtwopjbqhtts8seven
c138
mnrnhqtzthreesnnj2
81
qscxbblvsxsgtcpr4cphmxgmbrb
4six2mrvdpxqvhngrrzhjdjfp
4tmrlmdfivefivethree7
363
8488one
827pcxnhpk
ckbdqdvhdd4hrb9nnghltwo
5onecjjfdfrxppqkhrpthree5lr4
62lvlsp2sixfive5
czxfkhcbvcdhdzx2
jzdgngb68
fourtwo18pxkttcr8four
bkttjvvmxvzkhmxtpqrxxhslstwo6five22
four26mcvb2fivepone
dzptwo5ninejgjrfour
nrhqqnjfour9four
ffoneightpxp41six
threeninehzmrh8dzvmpgrhsv
5onefour5
jdl5fourtwo8
threefour8seventwo1seven
lvkjd6five6fxkfthxkgmfour6
three3seventhreepbsgc
7threethree6fgmfgcv6glrk
nine3five
54one9
jhvqhrvqfrfgpdrh3three5
3psztbfour4
nlffive1nxcqpfour232
3threetwosgsbdjnt
dtfstb1qhggj
qnnp9twojdg4kfzvtbsevenfour
qfnmdkfkpndd36sixsixfourseven
8seven7oneseven9x4five
4847onetgmsbjjqhtfiveksmmlftk
fivesixlklrnhtdfg3
cpcqone1pnxkhrkdhhzheightlpb5three
eighttwo6
seven2fourfourvfzxqbh7
frf1fivesix3snnlpgpc7
svtvfcsevenhszznbzpfiveeight2
lqpqzthree5mcmzpqlxkgpssix
9sltfzqsczpjbndmqbccgmq
5six9fourxppzkgbj6
8dqhklnxseven1
3two4
ktvlhmq3xzmcztbplxlqzpqmoneightffd
76dklx7two8nine5vgvh
xmgjgfz4
two7tcnceight9
ninecvfgpbrqcgvplt32six
g3rcbzqxnjllqoneklpbjzbqbseven
svkf8ggdnmeighthhjlz3four
hfdb44zqjmxbrft3qrpgvjngc1
dzsbnsrg35tm
qchpdcg2
9775fcg91
gdrhdgmfdk77eight
14bsixthreeseven
two1twoninepkkvhlzlt3
kvslhphhj9fourvljqbfive
8nc6
qg7six
7vkgv8
td5fivefive
xvfour8jttrsvvjkkdxxzjsl
7qbgdfkj4fourfivenmqtfmgeight
eightfourdzxpfggzxcqrmtxnggchlqtwokcnrdhqf1
lbhcptlfhflgthreefive9cpprdlbv
nine7941cfxnxszfivebm
ninenhgpj7fourfour9
456xg
ninethreesqdp11four32lgdl
38trrgt7
tnhhxlm3rvzvxmmcgpxm
oneqtmg25snbgzfhtkq8hdlvszrclcsrhqsdn
3cjndzmkfx
clxvdv7seven
five6three1
8sixfour36seven9mlxmrx7
vdrk27jnrbrvdrnine2nvnlkcszoneightchk
67drgjqrqdktplq
36pprvngxrrjtnpffmrxxdl
hkkrgxtpvzzmcrfcbcjzbpnvls88hrxl
b4jkeightsevenmxkdpvdpt857
6eighteight2four
eights73five
fiveslpjvzjlpvzone41nine
smnmlrxx7
onesixfive4
8s75eightone
1fivecpqsxm
99743ninesix
twopzflcvxsgzn1v8
9three49
two2sevenf8sixvdpxzgxbmkrz2
sevenfive6shhbqhgqqhstdrnfxsjtjvnine
kzqsvn29s
threenlrrhz4ttqljvr
qx9sfdx
nhsbgsvvp7
876eightfivetprjjbxhzbfpqgltt1
six3xdjtphrm1pbjvjfn
nflcpmjpt6six9two6xvbmbm
fmlglbgmhvseven4nnkkgcmbv9chmthxvztgsevenbtzs
7bkmchf4five
xgglktnbeight183twonine
onecbpdv9
fhhmddrnlnkvlb5drtcbrjprzcqlj7mgspc
7threexkfjklvzcfourtnrgmpveightfour8
6rhmcblhfourtwo3two
one4fiveonenine
slksfchvlgblninefszfbksddgvd4
four742sbseven3lk
8qbchrfldtvppv6fourzfivez
ninephsmnnvsfour1
one5kvcrtnc5sixonehjfxnsixfqhzjj
8drgrzngh
16eight8rczbjmblhn
threejpgjbpfdbvgdpftwo96
sfkhvjgd2smgfkr9two27
threeqpdeight3
threesevenxblrmqkvzvrbhsix4
9seven2vlhsninenvptdvmrpzbrlpfhfnl
xkjpdpgsrtvlfg1eight
seven254
three8fnmcfhrdtjdlpfqlqhgx
21bkjv54s8
82ddfour6vcggxhlmpfivesix
cdsbgtpqthreefivel8nine
5five5cksct7bvjbtdjkcz
64fhm
tqk5
3mhzkhtfzscpgfctfskdn2five
cvmvthreez8fivegvcbsxj35nfxdshqz
sevenqlqmkhdjpmsjrtggqzfive35scmnfpmhmp
sixfive2
6sevensvqhhfivetwo
eightthree84
82six8
8bbhrkzgtg59cpgdvb
11three3hfjkvmsl
qdljkm792
eightgq5xgskvtpgmhxsnhzlqhnksxflvx
ghdhv1seventhree55fourthreemrlhq
oneonethree3zm
3twobsxqfsfnine
eightsbznkpzsqqtpxfhlfr5
sevenchs2twoeight
ninezjtxp7bpzdgtoneeightoneighth
fmkbvzzgvl46fiveoneightqn
kvsonevxtmmxnfour4lpxjrbcfoursixp
391onelktkfd1three2twonels
pn2
3cmsmfvjchflngkrbtsfrblltwo
eightfour8ninesix29
dpdvnonefour2eightbzzrvthmkgfcdv5
59sfiverboneone3
6xrqq1
9bczfbfbhkntmdmseven68
npqm35qszmptx
fivepkdlmonefivesixz252
dzrtwonetbbv9twovrsp
fpkfourfivendkhz6five
86dknvgcnmcslmztwo3
fivetxrrzjpb6tqseightjthnhn
362zhtxdgvcdgnzv55
eightsevenbllbgvdonethreeprglpjhxhqfdn6csnghrtl
5fourc
8fournine3svdlh5sevenoneighttsq
eightjkqmgljpprcpblvrthreeseven8fourone7
vqrqlhmb9seven231vbfzrrstwohjvdjgg
kk1foursixeight849
seven13ttbnlr21
jfgsksrrkjsixsixthreesixfourrbxqzdhc9
3fiveeightqbndjdg6pmftqshone
nineoneseventkdpztbkk83two4
cjdbxhpzr87eight
twosevenkkdhf86sevendzhvkrttjljr
67cdltmnbkktzseightsix7two
three44zqtglqcffrkmhsgjfive
twoonenbf74three9tthsvmngcx
lvrnpvmvgvnnonesix5
nchcfxxphmnqbhfourxnine52sixj
fourqpvngjszftg88seven
xgzmv133zvjmlfqhcmjbqzfntkjtpdjlkmv
fivejxc4threebzqbhrldsf
glvtpvnsevenqvrtcgjkhkcsix1seven
sjeightwo4fivesevensixnqxsjckfive
z5
jxxhsmdvkglljhxzxljqz9zplmslcfdxxljtwosixkjxlsrfcx
fiversdslrkcqclzb7zdzlbllfkvphfqhnfour
five1hpdncfpzhdxrhnv4lblgthree7snj
8scqf
threepcfkrhstmsbfvqxsxrctwo185six
7lqvpqxbjrv9nine985
eight16jvfqcgbmhdgdltxz9
three8vncqf9
zrzp9sevenxthpsxtdninebmnhgrkdjqqjfffourblbkch
fiveseven25snpsnjgnfl
ninefourxrtdlb1threefivefive1rbqzfbxtst
three8three7five41
tcnvsevenz45fngtjljh
txvpst1
eighthrlbmtk4nssknqmxjvjnqsqlfivekzrphrtwo
two492ninekcjvllckdtkp
48vlvmkcvbsfoursix8
eight28sixsix3nine45
jkm8hhqfclneightjlpsnck8dcvlvgkrfprflsvlh
4zxjqlfn
fivebpqfjtcmlsixfour48onejqhzgtftrvkknngf
dkfour7fivezzsjdbnvgninefivenine5
twoxjqldtk91eightneight
seven57177
gpbqvlpxvnzgninefoureightthree5
bzprbrninesnjpgqthreeeightsixtgmvrthfour4
83brtxx4zzxsxrcfivebzbxlsl
threesevenninetwo8twoone
veightwochrmfrkrcppxkj4tvvzmhqjhnjvtq1threegtmfnnjpxb
3zgqpvq9krslgjgkjmtbsbhbjg
4nine8mhfs67vhdjvsjrlz2
nine659xrvhhnnbqq7
bninetwoeight8fxntpdkrfp9eightnb
qtwonecvbgxqfiveoneeight4five
seven926
shkkksn4onefivethree
eightsevenqvfqmzqsfn1
sixdgjmtk5fivevqbxxdzrjqdtgnnffzmfqxzjdzhp
stsjcmzbxt1
rvnone3three9twozchp26
ninefmpkdlvzfivetwo1
four4one4vdpgp9lprtqjzmhk
15rpdhczznrzclfcsxdx41
seven36twopnvjdxrss5
46ninenine
6jg8xtqfmjffdlfhzrn
2jsixfive573
two8eight39fbqkdzseven9
3gnrg5
hlkk1
1sevenfive7
72jvdqmlrddk
ninetwo8hzdninesixnjhgrcsttwo
23cxdccvvjrjfsixlcrmxjkeight
6brqjfjsix56nine3
ninezveighttwogc5two
ninexskpsth5sevennine
c7jmmjcv
89sixthreeeight71q
fivekccfngmxseven7onernfskqnqfive5
sevensixone4hfhgs
xq2jxddpdm3nine8ptqlm4
smsevendj19fbqdjpznfourgqq6
7147
znqbfmcpd4twoseven
fhcsmxtdx6
sxj2tcb92
6nine6
qdfknmdzt88
66threebsndqlkrshfive5
eight7pckdhlqzrx24
3ninevk
sevennine3
8one1rjtnhjx
eightthree3ninekzhtlqsevenssprmrqhhgncrs
6one8nlzxfxvr
4cgm9fivethree
four77gxvdqztzzgbsxhntwortndzqzj
t8three335
c2rjggzl
jjgjbqgbnz4gdsqk66991
ninezckbpsr9
jb5sevenseven
one5nfdcvx
1kbcmclhrh1onejzft
fiveninefivedglztnjxblonehfive3
6onesixh6onethree9
spdzhnt5tpzrkh1fxlnine4skgzdln
1onenineqgzcq2eightwonh
vfzvds826vtlrcg6rvseven
vqmoneight9tknqtcsmb
kqrcrqrqjbdeight7ckhr23
oneeight2
8eightnhtqcggtxc6dfsfcjfpznmsthree
sxfvfdkff8dvlmbdktsixmzpnxzmml2
9lgmxktj1frxl"""

In [2]:
# algorithm for solving the puzzle

def solve_calibration_problem(calibration_input):

    res_list_of_numbers = []

    # look at each line
    for line in calibration_input.splitlines():

        # extract first and last number
        first_number = None
        last_number = None

        # get first number
        for char in line:
            if char.isdigit():
                first_number = int(char)
                break

        # get second number (by itterating the reversed line)
        for char in line[::-1]:
            if char.isdigit():
                last_number = int(char)
                break

        # combine numbers to one integer
        line_value = first_number * 10 + last_number

        # save result in list
        res_list_of_numbers.append(line_value)

    # add all integers together
    final_res = sum(res_list_of_numbers)

    return final_res

In [3]:
# solve puzzle
solve_calibration_problem(personal_puzzle_input)

54081

The solution: 54081 is correct! :) 

--- Part Two ---
Your calculation isn't quite right. It looks like some of the digits are actually spelled out with letters: one, two, three, four, five, six, seven, eight, and nine also count as valid "digits".

Equipped with this new information, you now need to find the real first and last digit on each line. For example:

two1nine
eightwothree
abcone2threexyz
xtwone3four
4nineeightseven2
zoneight234
7pqrstsixteen

In this example, the calibration values are 29, 83, 13, 24, 42, 14, and 76. Adding these together produces 281.

What is the sum of all of the calibration values?

In [4]:
# input 

test_input = """two1nine
eightwothree
abcone2threexyz
xtwone3four
4nineeightseven2
zoneight234
7pqrstsixteen"""

In [47]:
# algorithm for solving the puzzle
def alter_input_for_second_part(calibration_input):
    
    def check_if_number(string_with_possible_number):
        # the function takes a string and checks if it contains a one digit integer written out
        # one digit integers can be represented between 3 to 5 char strings (one, two, six and three, seven, eight)
        # therefor the input string will be cut to 5 chars and checked for each possible value for zero to nine
        # this function return the number as a singel char and the length of the number as string
        
        # check if char is even long enought to contain a number
        if len(string_with_possible_number) > 2:
        
            # check for special cases where numbers are intertwined
            if len(string_with_possible_number) > 4:
                char_with_len_five = string_with_possible_number[:5]

                if char_with_len_five == "twone": return ("21", 5)

            if len(string_with_possible_number) > 6:
                char_with_len_seven = string_with_possible_number[:7]

                if char_with_len_seven == "oneight": return ("18", 7)
                elif char_with_len_seven == "eightwo": return ("82", 7)

            if len(string_with_possible_number) > 7:
                char_with_len_eight = string_with_possible_number[:8]

                if char_with_len_eight == "fiveight": return ("58", 8)
                elif char_with_len_eight == "nineight": return ("98", 8)

            if len(string_with_possible_number) > 8:
                char_with_len_nine = string_with_possible_number[:9]

                if char_with_len_nine == "sevenine": return ("79", 9)

            if len(string_with_possible_number) > 9:
                char_with_len_ten = string_with_possible_number[:10]

                if char_with_len_ten == "threeight": return ("38", 10)

            # check if char is even long enought to contain a number
            if len(string_with_possible_number) > 2:
                char_with_len_three = string_with_possible_number[:3]

                if char_with_len_three == "one": return ("1", 3)
                elif char_with_len_three == "two": return ("2", 3)
                elif char_with_len_three == "six": return ("6", 3)

            # check if char is long enough for numbers with 4 chars
            if len(string_with_possible_number) > 3:
                char_with_len_four = string_with_possible_number[:4]

                if char_with_len_four == "four": return ("4", 4)
                elif char_with_len_four == "five": return ("5", 4)
                elif char_with_len_four == "nine": return ("9", 4)

            # check if char is long enough for numbers with 5 chars
            if len(string_with_possible_number) > 4:
                char_with_len_five = string_with_possible_number[:5]

                if char_with_len_five == "three": return ("3", 5)
                elif char_with_len_five == "seven": return ("7", 5)
                elif char_with_len_five == "eight": return ("8", 5)
            
        return None
    
    def replace_char_in_string(line, number_as_singel_char, length_of_number_as_char, index_of_first_letter_of_number):

        left = line[:index_of_first_letter_of_number]
        mid = number_as_singel_char
        end = line[index_of_first_letter_of_number + length_of_number_as_char:]

        new_line = left + mid + end
        
        return new_line
    
    
    new_input_list = []
    
    # go through each line
    for line in calibration_input.splitlines():
        
        # variables to calculate remaining chars in the while loop
        length_of_line = len(line)
        index = 0
        remaining_length_of_line = length_of_line - index
        
        # for each index in line
        while index < length_of_line:
            
            # check if current char is beginning of a number
            possible_number_as_char_and_length = check_if_number(line[index:])
            
            # if yes, replace string number with singel char number in line string
            if possible_number_as_char_and_length is not None:
                line = replace_char_in_string(line, possible_number_as_char_and_length[0], 
                                                  possible_number_as_char_and_length[1], index)
                
                # update variables for new line
                length_of_line = len(line)
            
            index = index + 1
            remaining_length_of_line = length_of_line - index
                                              
        new_input_list.append(line)
        
        # convert list to sting with line breaks
        new_input = '\n'.join(new_input_list)
        
    return new_input

In [51]:
def solve_calibration_problem_part_two(calibration_input):
    
    # alter input
    new_input = alter_input_for_second_part(calibration_input)
    
    # solve with original function
    res = solve_calibration_problem(new_input)
    
    return res

In [53]:
solve_calibration_problem_part_two(personal_puzzle_input)

54649

The solution: 54649 is correct! :)